opentripplanner/otp-react-redux
# chore(lint): Add pre-commit hook to enforce formatting.

This PR adds a pre-commit hook to run `eslint` on files specified using `git add`. `mastarm` rules are used, ~plus additional rules regarding sorting of prop keys and variable declarations~ (see also https://github.com/conveyal/mastarm/pull/310).

Assumptions:
- You have eslint installed system-wide (otherwise, probably run `yarn global add eslint`)
- ~You have cloned the [mastarm repo](https://github.com/conveyal/mastarm), and repos for your `mastarm` and `otp-react-redux` are in the same directory.~ => Not necessary, the PR uses the existing mastarm dependency.

To test this PR after checkout:
1. Run `yarn install`
2. Add blank spaces/lines to a `.js` file, ~change the order of props or var declarations,~ etc.
3. Commit the changes
4. Observe that your extra spaces ~and prop disordering~ have been fixed, e.g.
![image](https://user-images.githubusercontent.com/56846598/120685124-82362d00-c46d-11eb-9471-eeca56c33773.png)
5. Run `git reset HEAD~` to undo your commit and undo your file changes.

This PR uses a version of husky that is more recent than in OTP-UI, and the way it is configured is thus different. Specifically, husky introduces:
- hook files and its own config directory
- a `postinstall` script that is ran after install, except when `prepublish` is involved. For `prepublish` steps, the library authors recommend disabling installing husky hooks (see [these docs](https://typicode.github.io/husky/#/?id=install-1)).
